Content.Shared/Anomaly/Components/InnerBodyAnomalyComponent.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Content.Shared.Anomaly.Effects;
+using Content.Shared.Body.Prototypes;
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Anomaly.Components;
+
+/// <summary>
+/// An anomaly within the body of a living being. Controls the ability to return to the standard state.
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState(true), Access(typeof(SharedInnerBodyAnomalySystem))]
+public sealed partial class InnerBodyAnomalyComponent : Component
+{
+    [DataField]
+    public bool Injected;
+
+    /// <summary>
+    /// A prototype of an entity whose components will be added to the anomaly host **AND** then removed at the right time
+    /// </summary>
+    [DataField(required: true)]
+    public EntProtoId? InjectionProto;
+
+    /// <summary>
+    /// Duration of stun from the effect of the anomaly
+    /// </summary>
+    [DataField]
+    public float StunDuration = 4f;
+
+    /// <summary>
+    /// A message sent in chat to a player who has become infected by an anomaly
+    /// </summary>
+    [DataField]
+    public LocId? StartMessage = null;
+
+    /// <summary>
+    /// A message sent in chat to a player who has cleared an anomaly
+    /// </summary>
+    [DataField]
+    public LocId? EndMessage = "inner-anomaly-end-message";
+
+    /// <summary>
+    /// Sound, playing on becoming anomaly
+    /// </summary>
+    [DataField]
+    public SoundSpecifier? StartSound = new SoundPathSpecifier("/Audio/Effects/inneranomaly.ogg");
+
+    /// <summary>
+    /// Action in access of an entity
+    /// </summary>
+    [DataField]
+    public EntityUid? Action = null;
+
```

Comment:
ActionGrantComponent can apply this instead and you can remove the corresponding code.